docs/consumer.adoc

```diff
@@ -0,0 +1,277 @@
+
+First thing to do is add typical Arquillian dependencies such as Arquillian bom and the dependency of the container you want to use.
+Nothing new here:
+
+[[arqdeps]]
+[source, xml]
+.pom.xml
+----
+<dependencyManagement>
+    <dependencies>
+        <dependency>
+            <groupId>org.jboss.arquillian</groupId>
+            <artifactId>arquillian-bom</artifactId>
+            <version>${version.arquillian_core}</version>
+            <type>pom</type>
+            <scope>import</scope>
+        </dependency>
+    </dependencies>
+</dependencyManagement>
+
+<dependencies>
+    <dependency>
+        <groupId>org.jboss.arquillian.junit</groupId>
+        <artifactId>arquillian-junit-container</artifactId>
+        <scope>test</scope>
+    </dependency>
+    <dependency>
+        <groupId>org.arquillian.container</groupId>
+        <artifactId>arquillian-container-chameleon</artifactId> <!--1-->
+        <version>1.0.0.Alpha7</version>
+        <scope>test</scope>
+    </dependency>
+</dependencies>
+----
+<1> Notice that I am using Chameleon generic container, you can read more http://www.lordofthejars.com/2016/09/arquillian-chameleon-for-sake-of.html[here].
+
+Then you need to add `arquillian-pact-consumer` dependency as well as `pact-jvm-consumer` dependency:
+
+[source, xml]
+----
+<dependencies>
+    <dependency>
+        <groupId>org.arquillian.algeron</groupId>
+        <artifactId>arquillian-algeron-pact-consumer-core</artifactId>
+        <version>${version.arquillian_algeron}</version>
+        <scope>test</scope>
+    </dependency>
+    <dependency>
+        <groupId>au.com.dius</groupId>
+        <artifactId>pact-jvm-consumer_2.11</artifactId>
+        <scope>test</scope>
+        <version>3.5.0-beta.2</version> <!--1-->
+    </dependency>
+</dependencies>
+----
+<1> Arquillian Algeron Pact has been tested with latest version, but it should work with any 3.X version.
+
+After dependencies you can write the test that defines the contract:
+
+
+[[incontainer]]
+[source, java]
+.ClientGatewayTest.java
+----
+@RunWith(Arquillian.class) // <1>
+@Pact(provider="test_provider", consumer="test_consumer") // <2>
+public class ClientGatewayTest {
+
+    @Deployment // <3>
+    public static JavaArchive createDeployment() {
+        return ShrinkWrap.create(JavaArchive.class).addClasses(ClientGateway.class);
+    }
+
+    public PactFragment createFragment(PactDslWithProvider builder) {
+
+        Map<String, String> header = new HashMap<>();
+        header.put("Content-Type", "application/json");
+
+        return builder
+                .given("test state")
+                .uponReceiving("ConsumerTest test interaction")
+                .path("/")
+                .method("GET")
+                .willRespondWith()
+                .status(200)
+                .headers(header)
+                .body("{\"responsetest\": true, \"name\": \"harry\"}")
+                .toFragment(); // <4>
+    }
+
+    @EJB // <5>
```

Comment:
Just a personal note - I wouldn't use `@EJB` - it's not considered as cool technology anymore :)